From 8584c3d462e501b675442be5ccd9387f1b3dcf42 Mon Sep 17 00:00:00 2001
From: Wayne Lee <waynelee@silklabs.com>
Date: Thu, 24 Sep 2015 14:47:33 -0700
Subject: [PATCH 3/3] Add support for XCode 7

---
 core/combo/mac_version.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/combo/mac_version.mk b/core/combo/mac_version.mk
index b49feee..7b37c3a 100644
--- a/core/combo/mac_version.mk
+++ b/core/combo/mac_version.mk
@@ -9,7 +9,7 @@ ifndef build_mac_version
 
 build_mac_version := $(shell sw_vers -productVersion)
 
-mac_sdk_versions_supported :=  10.6 10.7 10.8 10.9
+mac_sdk_versions_supported :=  10.6 10.7 10.8 10.9 10.11
 ifneq ($(strip $(MAC_SDK_VERSION)),)
 mac_sdk_version := $(MAC_SDK_VERSION)
 ifeq ($(filter $(mac_sdk_version),$(mac_sdk_versions_supported)),)
-- 
2.5.0

