From 8af5608911f63b119672527385ca3004a4dcdce2 Mon Sep 17 00:00:00 2001
From: Michael Vines <mvines@silklabs.com>
Date: Mon, 2 Jan 2017 18:25:09 -0800
Subject: [PATCH] Remove APN list

---
 common.mk | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/common.mk b/common.mk
index 48b3ea5..04a28ff 100644
--- a/common.mk
+++ b/common.mk
@@ -160,11 +160,6 @@ PRODUCT_PACKAGES += \
 PRODUCT_PACKAGES += \
     fs_config_files
 
-# APN list
-PRODUCT_COPY_FILES += \
-    device/sample/etc/old-apns-conf.xml:system/etc/old-apns-conf.xml \
-    device/sample/etc/apns-full-conf.xml:system/etc/apns-conf.xml
-
 # Limit dex2oat threads to improve thermals
 PRODUCT_PROPERTY_OVERRIDES += \
     dalvik.vm.dex2oat-threads=2 \
-- 
2.10.0

