From d6d0e367701f28eb232f8119563e51e04d86f3f1 Mon Sep 17 00:00:00 2001
From: Michael Vines <mvines@silklabs.com>
Date: Wed, 1 Jul 2015 16:29:53 -0700
Subject: [PATCH] sdcard: remove derived permissions

---
 init.shamu.rc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/init.shamu.rc b/init.shamu.rc
index f2a060b..8967337 100644
--- a/init.shamu.rc
+++ b/init.shamu.rc
@@ -500,7 +500,7 @@ on property:ro.data.large_tcp_window_size=true
     write /proc/sys/net/ipv4/tcp_adv_win_scale  2
 
 # virtual sdcard daemon running as media_rw (1023)
-service sdcard /system/bin/sdcard -u 1023 -g 1023 -l /data/media /mnt/shell/emulated
+service sdcard /system/bin/sdcard -u 1023 -g 1023 /data/media /mnt/shell/emulated
     class late_start
 
 on charger
-- 
1.9.1

