From 00a32a08cc8ac67531e621903cd933a3a814810b Mon Sep 17 00:00:00 2001
From: Chris Jones <chris@silklabs.com>
Date: Thu, 18 Jun 2015 08:58:11 -0700
Subject: [PATCH] Add silk product

---
 aosp_shamu.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/aosp_shamu.mk b/aosp_shamu.mk
index e5fa301..dc8c55a 100644
--- a/aosp_shamu.mk
+++ b/aosp_shamu.mk
@@ -28,8 +28,8 @@ PRODUCT_DEVICE := shamu
 PRODUCT_BRAND := Android
 PRODUCT_MODEL := AOSP on Shamu
 PRODUCT_MANUFACTURER := motorola
-PRODUCT_RESTRICT_VENDOR_FILES := true
 
+$(call inherit-product, product/$(SILK_PRODUCT)/$(SILK_PRODUCT).mk)
 $(call inherit-product, device/moto/shamu/device.mk)
 $(call inherit-product-if-exists, vendor/moto/shamu/device-vendor.mk)
 
-- 
1.9.1

