From 945fe8303fbf8af9cda387b9352c1ca2a82581de Mon Sep 17 00:00:00 2001
From: Michael Vines <mvines@silklabs.com>
Date: Tue, 10 Nov 2015 15:28:19 -0800
Subject: [PATCH] Use atomic-polyfill for Android/libstlport

---
 src/node.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/node.cc b/src/node.cc
index cf2b501..6e74497 100644
--- a/src/node.cc
+++ b/src/node.cc
@@ -86,7 +86,7 @@ typedef int mode_t;
 extern char **environ;
 #endif
 
-#ifdef __APPLE__
+#if defined(__APPLE__) || defined(__ANDROID__)
 #include "atomic-polyfill.h"  // NOLINT(build/include_order)
 namespace node { template <typename T> using atomic = nonstd::atomic<T>; }
 #else
-- 
1.9.1

