From d4dcaf7eb9d10c05029299e6fa8e7e2a853fe6d9 Mon Sep 17 00:00:00 2001
From: Michael Vines <mvines@silklabs.com>
Date: Wed, 26 Oct 2016 14:44:35 -0700
Subject: [PATCH] Turn off GTEST c++11 extensions

---
 common.gypi | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/common.gypi b/common.gypi
index 01ab2de..8df8617 100644
--- a/common.gypi
+++ b/common.gypi
@@ -332,7 +332,7 @@
       ['OS=="android"', {
         'target_conditions': [
           ['_toolset=="target"', {
-            'defines': [ '_GLIBCXX_USE_C99_MATH' ],
+            'defines': [ '_GLIBCXX_USE_C99_MATH', 'GTEST_LANG_CXX11=0' ],
             'libraries': [ '-llog' ],
           }],
         ],
-- 
2.10.0

