From b242c0bc8d273bbc1f18fa050c052ddc4c03f1b6 Mon Sep 17 00:00:00 2001
From: Michael Vines <mvines@silklabs.com>
Date: Wed, 26 Oct 2016 17:28:29 -0700
Subject: [PATCH 03/11] De-pie. This breaks linking native .node modules

---
 common.gypi | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/common.gypi b/common.gypi
index af9c6bafb0..795624ee8e 100644
--- a/common.gypi
+++ b/common.gypi
@@ -76,10 +76,6 @@
             'cflags': [ '-gxcoff' ],
             'ldflags': [ '-Wl,-bbigtoc' ],
           }],
-          ['OS == "android"', {
-            'cflags': [ '-fPIE' ],
-            'ldflags': [ '-fPIE', '-pie' ]
-          }],
           ['node_shared=="true"', {
             'msvs_settings': {
              'VCCLCompilerTool': {
@@ -126,10 +122,6 @@
           ['OS!="mac" and OS!="win"', {
             'cflags': [ '-fno-omit-frame-pointer' ],
           }],
-          ['OS == "android"', {
-            'cflags': [ '-fPIE' ],
-            'ldflags': [ '-fPIE', '-pie' ]
-          }],
           ['node_shared=="true"', {
             'msvs_settings': {
              'VCCLCompilerTool': {
-- 
2.12.3

