From 0610de6bef61593e84fca0a4c19adfc2f2449607 Mon Sep 17 00:00:00 2001
From: Michael Vines <mvines@silklabs.com>
Date: Wed, 12 Apr 2017 09:49:08 -0700
Subject: [PATCH 05/11] Enable process user/group methods

---
 src/node.cc | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/src/node.cc b/src/node.cc
index 6ca8ee95e2..6a7b97ff37 100644
--- a/src/node.cc
+++ b/src/node.cc
@@ -83,7 +83,7 @@ typedef int mode_t;
 #include <unistd.h>  // setuid, getuid
 #endif
 
-#if defined(__POSIX__) && !defined(__ANDROID__)
+#if defined(__POSIX__)
 #include <pwd.h>  // getpwnam()
 #include <grp.h>  // getgrnam()
 #endif
@@ -1922,7 +1922,7 @@ static void Umask(const FunctionCallbackInfo<Value>& args) {
 }
 
 
-#if defined(__POSIX__) && !defined(__ANDROID__)
+#if defined(__POSIX__)
 
 static const uid_t uid_not_found = static_cast<uid_t>(-1);
 static const gid_t gid_not_found = static_cast<gid_t>(-1);
@@ -1974,7 +1974,8 @@ static gid_t gid_by_name(const char* name) {
   errno = 0;
   pp = nullptr;
 
-  if (getgrnam_r(name, &pwd, buf, sizeof(buf), &pp) == 0 && pp != nullptr) {
+  pp = getgrnam(name);
+  if (pp != nullptr) {
     return pp->gr_gid;
   }
 
@@ -3398,7 +3399,7 @@ void SetupProcessObject(Environment* env,
 
   env->SetMethod(process, "umask", Umask);
 
-#if defined(__POSIX__) && !defined(__ANDROID__)
+#if defined(__POSIX__)
   env->SetMethod(process, "getuid", GetUid);
   env->SetMethod(process, "geteuid", GetEUid);
   env->SetMethod(process, "setuid", SetUid);
-- 
2.12.3

