From 73afaf9e8dc460286c6ff902c875e32593e28e1e Mon Sep 17 00:00:00 2001
From: Michael Vines <mvines@silklabs.com>
Date: Wed, 26 Oct 2016 14:43:08 -0700
Subject: [PATCH 08/11] <type_traits> in some gonks need help finding
 std::size_t

---
 src/util.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/util.h b/src/util.h
index f5b23e9c5b..f984695f33 100644
--- a/src/util.h
+++ b/src/util.h
@@ -19,6 +19,7 @@
 #ifdef USE_TR1_TYPE_TRAITS
 #include <tr1/type_traits>  // NOLINT(build/c++tr1)
 #else
+#include <cstddef>
 #include <type_traits>  // std::remove_reference
 #endif
 
-- 
2.12.3

