From e900db51ffa07f9a9bf9ad1d728d24a00100df6e Mon Sep 17 00:00:00 2001
From: Chris Jones <cjones.bugs@gmail.com>
Date: Mon, 25 May 2015 17:03:17 -0700
Subject: [PATCH 09/11] Don't allow passing AI_V4MAPPED to bionic
 |getaddrinfo()|.

---
 src/cares_wrap.cc | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/src/cares_wrap.cc b/src/cares_wrap.cc
index 0a0aff95e2..a71aa7da06 100644
--- a/src/cares_wrap.cc
+++ b/src/cares_wrap.cc
@@ -1367,7 +1367,16 @@ static void Initialize(Local<Object> target,
   target->Set(FIXED_ONE_BYTE_STRING(env->isolate(), "AI_ADDRCONFIG"),
               Integer::New(env->isolate(), AI_ADDRCONFIG));
   target->Set(FIXED_ONE_BYTE_STRING(env->isolate(), "AI_V4MAPPED"),
-              Integer::New(env->isolate(), AI_V4MAPPED));
+              Integer::New(env->isolate(),
+#ifdef __ANDROID__
+                           // Android's wonderful little short-bus
+                           // libc doesn't understand AI_V4MAPPED or
+                           // AI_V4MAPPED_CFG.
+                           0
+#else
+                           AI_V4MAPPED
+#endif
+                  ));
 
   Local<FunctionTemplate> aiw =
       FunctionTemplate::New(env->isolate(), NewGetAddrInfoReqWrap);
-- 
2.12.3

