From 6211dfce5fd99c3535932d50febd41836655de97 Mon Sep 17 00:00:00 2001
From: Michael Vines <mvines@silklabs.com>
Date: Wed, 26 Oct 2016 14:44:35 -0700
Subject: [PATCH 10/11] Turn off GTEST c++11 extensions

---
 common.gypi | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/common.gypi b/common.gypi
index 795624ee8e..afcc8acfaf 100644
--- a/common.gypi
+++ b/common.gypi
@@ -331,7 +331,7 @@
       ['OS=="android"', {
         'target_conditions': [
           ['_toolset=="target"', {
-            'defines': [ '_GLIBCXX_USE_C99_MATH' ],
+            'defines': [ '_GLIBCXX_USE_C99_MATH', 'GTEST_LANG_CXX11=0' ],
             'libraries': [ '-llog' ],
           }],
         ],
-- 
2.12.3

