From dc6986f53f460a32c572ef9f3fb0005db68aac64 Mon Sep 17 00:00:00 2001
From: Michael Vines <mvines@silklabs.com>
Date: Fri, 12 Feb 2016 23:14:13 -0800
Subject: [PATCH] Remove librecovery_testdata, causing trouble on M

---
 Android.mk | 14 --------------
 1 file changed, 14 deletions(-)

diff --git a/Android.mk b/Android.mk
index 6a0c689..e17bbf8 100644
--- a/Android.mk
+++ b/Android.mk
@@ -62,18 +62,4 @@ LOCAL_CFLAGS := $(LIBRECOVERY_CFLAGS)
 LOCAL_MODULE_TAGS := tests
 include $(BUILD_SILK_EXECUTABLE)
 
-# librecovery test data -- this uses a local make target instead of an AOSP
-# module to avoid being built by default (signing an update zip requires java)
-TESTS_DIR := $(LOCAL_PATH)/tests
-OUT_DIR := $(shell cd $(call local-intermediates-dir); pwd)
-
-.PHONY: librecovery_testdata
-librecovery_testdata: librecovery_test
-	mkdir -p $(OUT_DIR)
-	export TARGET_DEVICE="$(TARGET_DEVICE)" && \
-	export LIBRECOVERY_ENV_VARS="$(DEVICE_CONFIG)" && \
-	$(foreach env_var,$(LIBRECOVERY_ENV_VARS),export $(env_var) &&) \
-	$(foreach dirname,$(LIBRECOVERY_TESTDATA),$(TESTS_DIR)/create_update.sh $(TESTS_DIR)/data/$(dirname) $(OUT_DIR)/$(dirname).zip &&) \
-	echo
-
 endif # $(ENABLE_LIBRECOVERY) == true
-- 
1.9.1

