From dda36d872ef81e7dc689d3987fc49445ecc5803e Mon Sep 17 00:00:00 2001
From: Ben Turner <bent@silklabs.com>
Date: Wed, 16 Sep 2015 13:44:04 -0700
Subject: [PATCH] Add support for XCode 7

---
 core/combo/mac_version.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/combo/mac_version.mk b/core/combo/mac_version.mk
index 6defba7..7b37c3a 100644
--- a/core/combo/mac_version.mk
+++ b/core/combo/mac_version.mk
@@ -9,7 +9,7 @@ ifndef build_mac_version
 
 build_mac_version := $(shell sw_vers -productVersion)
 
-mac_sdk_versions_supported :=  10.6 10.7 10.8 10.9
+mac_sdk_versions_supported :=  10.6 10.7 10.8 10.9 10.11 10.12
 ifneq ($(strip $(MAC_SDK_VERSION)),)
 mac_sdk_version := $(MAC_SDK_VERSION)
 ifeq ($(filter $(mac_sdk_version),$(mac_sdk_versions_supported)),)
-- 
2.5.2

