From 8711c4d91b5961e776513a906ab67e30cf602db8 Mon Sep 17 00:00:00 2001
From: Michael Vines <mvines@silklabs.com>
Date: Fri, 29 Apr 2016 17:36:03 -0700
Subject: [PATCH] Add Silk project

Change-Id: I85412a5188f27c4cb8d9fcd55d72b8e743784221
---
 base.mk | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/base.mk b/base.mk
index f180b69..d8de9f0 100644
--- a/base.mk
+++ b/base.mk
@@ -846,3 +846,5 @@ ifeq ($(TARGET_BUILD_VARIANT),user)
 PRODUCT_DEFAULT_PROPERTY_OVERRIDES+= \
     ro.adb.secure=1
 endif
+
+$(call inherit-product, product/$(SILK_PRODUCT)/$(SILK_PRODUCT).mk)
-- 
1.9.1

