From f0a785ea7338dae0a16de71ca6da78e17b94e4e9 Mon Sep 17 00:00:00 2001
From: Michael Vines <mvines@silklabs.com>
Date: Thu, 29 Sep 2016 21:06:51 -0700
Subject: [PATCH] sdcard: remove derived permissions

---
 rootdir/init.common.rc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rootdir/init.common.rc b/rootdir/init.common.rc
index 012de48..98e9239 100644
--- a/rootdir/init.common.rc
+++ b/rootdir/init.common.rc
@@ -378,7 +378,7 @@ service iprenew_bt-pan /system/bin/dhcpcd -n
     disabled
     oneshot
 
-service sdcard /system/bin/sdcard -u 1023 -g 1023 -t 4 -l /data/media /mnt/shell/emulated
+service sdcard /system/bin/sdcard -u 1023 -g 1023 -t 4 /data/media /mnt/shell/emulated
     class late_start
 
 service fuse_sdcard1 /system/bin/sdcard -u 1023 -g 1023 -w 1023 -t 4 -d /mnt/media_rw/sdcard1 /storage/sdcard1
-- 
2.8.4 (Apple Git-73)

