From 05be7a8a7800f3897739d6085c856854ee0f0180 Mon Sep 17 00:00:00 2001
From: Michael Vines <mvines@silklabs.com>
Date: Sun, 3 Sep 2017 10:17:55 -0700
Subject: [PATCH] Drop unneeded CURDIR

---
 Android.mk | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Android.mk b/Android.mk
index 3f6ae37..80212bc 100644
--- a/Android.mk
+++ b/Android.mk
@@ -15,7 +15,7 @@ target_crtbegin_dynamic_o := $($(LOCAL_2ND_ARCH_VAR_PREFIX)TARGET_OUT_INTERMEDIA
 target_crtbegin_static_o := $($(LOCAL_2ND_ARCH_VAR_PREFIX)TARGET_OUT_INTERMEDIATE_LIBRARIES)/crtbegin_static.o
 target_crtend_o := $($(LOCAL_2ND_ARCH_VAR_PREFIX)TARGET_OUT_INTERMEDIATE_LIBRARIES)/crtend_android.o
 
-BB_LDFLAGS := -nostdlib -Bdynamic -Wl,-z,muldefs$(shell if test $(PLATFORM_SDK_VERSION) -lt 16; then echo -ne ',-T$(CURDIR)/$(BUILD_SYSTEM)/armelf.x'; fi),-dynamic-linker,$(LINKER_BIN),-z,nocopyreloc,--no-undefined $(CURDIR)/$(target_crtbegin_dynamic_o) $(CURDIR)/$(target_crtend_o) $(CURDIR)/$(TARGET_OUT_SHARED_LIBRARIES)/libc.so -L$(CURDIR)/$(TARGET_OUT_SHARED_LIBRARIES)
+BB_LDFLAGS := -nostdlib -Bdynamic -Wl,-z,muldefs$(shell if test $(PLATFORM_SDK_VERSION) -lt 16; then echo -ne ',-T$(BUILD_SYSTEM)/armelf.x'; fi),-dynamic-linker,$(LINKER_BIN),-z,nocopyreloc,--no-undefined $(target_crtbegin_dynamic_o) $(target_crtend_o) $(TARGET_OUT_SHARED_LIBRARIES)/libc.so -L$(TARGET_OUT_SHARED_LIBRARIES)
 # FIXME remove -fno-strict-aliasing once all aliasing violations are fixed
 BB_COMPILER_FLAGS := $(subst -I ,-I$(CURDIR)/,$(subst -include ,-include $(CURDIR)/,$(TARGET_GLOBAL_CFLAGS))) $(foreach d,$(TARGET_C_INCLUDES),-I$(CURDIR)/$(d)) -fno-stack-protector -Wno-error=format-security -fno-strict-aliasing
 # for O and master
@@ -33,7 +33,7 @@ include $(BUILD_PREBUILT)
 
 BB_OUT_INTERMEDIATES := $(dir $(LOCAL_BUILT_MODULE))
 
-BB_MAKE_FLAGS := O=$(CURDIR)/$(BB_OUT_INTERMEDIATES) CROSS_COMPILE=$(notdir $(TARGET_TOOLS_PREFIX)) CONFIG_EXTRA_CFLAGS="$(BB_COMPILER_FLAGS)" EXTRA_LDFLAGS="$(BB_LDFLAGS)" LDLIBS="$(BB_LDLIBS)" $(BB_VERBOSE)
+BB_MAKE_FLAGS := O=$(BB_OUT_INTERMEDIATES) CROSS_COMPILE=$(notdir $(TARGET_TOOLS_PREFIX)) CONFIG_EXTRA_CFLAGS="$(BB_COMPILER_FLAGS)" EXTRA_LDFLAGS="$(BB_LDFLAGS)" LDLIBS="$(BB_LDLIBS)" $(BB_VERBOSE)
 
 $(BB_OUT_INTERMEDIATES):
 	mkdir -p "$@"
-- 
2.11.0

