From 9c268e43411c06c2751413043df350cefec63fa1 Mon Sep 17 00:00:00 2001
From: Michael Vines <mvines@silklabs.com>
Date: Fri, 25 Aug 2017 15:52:20 -0700
Subject: [PATCH] Relax system checks

---
 core/main.mk | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/core/main.mk b/core/main.mk
index d466ba516..c7a6f9bea 100644
--- a/core/main.mk
+++ b/core/main.mk
@@ -160,7 +160,7 @@ $(warning ************************************************************)
 $(warning You are building on a case-insensitive filesystem.)
 $(warning Please move your source tree to a case-sensitive filesystem.)
 $(warning ************************************************************)
-$(error Case-insensitive filesystems not supported)
+$(warning Case-insensitive filesystems not supported)
 endif
 
 # Make sure that there are no spaces in the absolute path; the
@@ -207,7 +207,7 @@ $(info $(space))
 $(info Please follow the machine setup instructions at)
 $(info $(space)$(space)$(space)$(space)https://source.android.com/source/initializing.html)
 $(info ************************************************************)
-$(error stop)
+#$(error stop)
 endif
 
 # Check for the current JDK.
@@ -228,7 +228,7 @@ $(info ************************************************************)
 $(info You asked for an OpenJDK based build but your version is)
 $(info $(java_version_str).)
 $(info ************************************************************)
-$(error stop)
+#$(error stop)
 endif # java version is not OpenJdk
 else # if requires_openjdk
 ifneq ($(shell echo '$(java_version_str)' | grep -i openjdk),)
@@ -239,7 +239,7 @@ $(info You use OpenJDK but only Sun/Oracle JDK is supported.)
 $(info Please follow the machine setup instructions at)
 $(info $(space)$(space)$(space)$(space)https://source.android.com/source/download.html)
 $(info ************************************************************)
-$(error stop)
+#$(error stop)
 endif # java version is not Sun Oracle JDK
 endif # if requires_openjdk
 
-- 
2.11.0

